File: trunk/example/vb/Filter2.vb
REM
REM  \file       example/vb/Filter2.vb
REM  \brief      \$Id: Filter2.vb 507 2009-11-28 15:18:46Z dev1usr $
REM  
REM  (C) 2009 - NHI - #1 - Project - Group
REM  
REM  \version    \$Rev: 507 $
REM  \author     EMail: aotto1968 at users.sourceforge.net
REM  \attention  this software has GPL permissions to copy
REM              please contact AUTHORS for additional information
REM


Imports System
Imports csmsgque

Public Module example
  Private Class Filter2
    Inherits MqS
    Implements IFilterFTR

    Private data As New List(Of List(Of String))

    ' service definition
    Public Sub FilterFTR() Implements csmsgque.IFilterFTR.Call
      Throw New ApplicationException("my error")
    End Sub
  End Class

  Sub Main(ByVal args() As String)
    Dim srv As New Filter2()
    Try
      srv.ConfigSetName("filter")
      srv.LinkCreate(args)
      srv.ProcessEvent(MqS.WAIT.FOREVER)
    Catch ex As Exception
      srv.ErrorSet(ex)
    Finally
      srv.Exit()
    End Try
  End Sub
End Module

File: trunk/example/java/Filter2.java
/**
 *  \file       example/java/Filter2.java
 *  \brief      \$Id: Filter2.java 507 2009-11-28 15:18:46Z dev1usr $
 *  
 *  (C) 2009 - NHI - #1 - Project - Group
 *  
 *  \version    \$Rev: 507 $
 *  \author     EMail: aotto1968 at users.sourceforge.net
 *  \attention  this software has GPL permissions to copy
 *              please contact AUTHORS for additional information
 */
package example;
import javamsgque.*;

class Filter2 extends MqS implements IFilterFTR {

  // service definition from IFilter
  public void FTR () {
    throw new RuntimeException("my error");
  }

  public static void main(String[] argv) {
    Filter2 srv = new Filter2();
    try {
      srv.ConfigSetName("filter");
      srv.LinkCreate(argv);
      srv.ProcessEvent(MqS.WAIT.FOREVER);
    } catch (Throwable e) {
      srv.ErrorSet(e);
    } finally {
      srv.Exit();
    }
  }
}
